Sources/StreamChat/Database/DTOs/ChannelDTO.swift

```diff
@@ -79,8 +79,7 @@ class ChannelDTO: NSManagedObject {
         
         // Update the date for sorting every time new message in this channel arrive.
         // This will ensure that the channel list is updated/sorted when new message arrives.
-        let lastDate = lastMessageAt ?? createdAt
```

Comment:
Doesn't this mean we will NOT always update the channel as it is stated in the comment? Is this to cause any issues?

Reply:
We need to check how it works with an empty channel. Previously we were making an assumption that `lastMessageAt ?? createdAt` always gives a non-optional date and `defaultSortingAt` is always has a value and can be used in sorting. Now, an empty channel can go to the bottom of the channel list when created

Reply:
Btw, if the sorting option is not provided, the backend returns channels sorted by `updatedAt` (not by `lastMessageAt`)

Reply:
Another issue is that we mark `createdAt/updatedAt` as non-optional fields but we don't give a default value. Can that be the root cause of the crash?

Reply:
> Yeah, we need to check how it works with an empty channel
It works

> Now, an empty channel can go to the bottom of the channel list when created
This is not happening

> Btw, if the sorting option is not provided, the backend returns channels sorted by updatedAt (not by lastMessageAt)
This issue is already there if it is true

> Another issue is that we mark createdAt/updatedAt as non-optional fields but don't give a default value. Can that be the root cause of the crash?
We could also do this yes. Actually, I think we should do it for all dates, to avoid this annoying crash.

Reply:
Can we check that providing default values for `createdAt/updatedAt` also fixes the crash?

Reply:
I will need to replicate this state again and confirm. I'll try to do this, but I'm not sure how I was able to reproduce this state. Will keep you posted 👍